src/API.jl

```diff
@@ -946,6 +946,20 @@ function precompile(ctx::Context; internal_call::Bool=false)
         was_recompiled[pkgid] = false
     end
     
+    # guarding against circular deps
+    function in_deps(pkg, deps, dmap)
+        length(deps) == 0 && return false
+        pkg in deps && return true
+        return any(map(dep->in_deps(pkg, dmap[dep], dmap), deps))
```

Comment:
```suggestion
        return any(dep->in_deps(pkg, dmap[dep], dmap)
```

Reply:
This suggestion doesn't seem right, and I'm not sure how to fix it.. 🤔

Reply:
```suggestion
        return any(dep->in_deps(pkg, dmap[dep], dmap), deps)
```

Reply:
sorry yeah, i was on my phone

Reply:
Ah, of course. Thanks